cmd/util/utils.go

```diff
@@ -24,39 +24,51 @@ var testNet3GenHash = Hash{
 	0x01, 0xea, 0x33, 0x09, 0x00, 0x00, 0x00, 0x00,
 }
 
+var regTestGenHash = Hash{
+  0x06, 0x22, 0x6e, 0x46, 0x11, 0x1a, 0x0b, 0x59,
+  0xca, 0xaf, 0x12, 0x60, 0x43, 0xeb, 0x5b, 0xbf,
+  0x28, 0xc3, 0x4f, 0x3a, 0x5e, 0x33, 0x2a, 0x1f,
+  0xc7, 0xb2, 0xb7, 0x3c, 0xf1, 0x88, 0x91, 0x0f,
+}
+
+func GenHashForNet(net wire.BitcoinNet) Hash {
+	switch net {
+		case wire.TestNet3:
+			return testNet3GenHash
+		case wire.MainNet:
+			return mainNetGenHash
+		case wire.TestNet: // yes, this is regtest
+			return regTestGenHash
+	}
+	panic("net not supported by GenHashForNet")
+}
+
 // Checks if the blk00000.dat file in the current directory
-// is testnet3 or mainnet.
-func CheckTestnet(isTestnet bool) {
-	if isTestnet == true {
-		f, err := os.Open("blk00000.dat")
-		if err != nil {
-			panic(err)
-		}
-		var magicbytes [4]byte
-		f.Read(magicbytes[:])
-
-		// Check if the magicbytes are for testnet
-		if magicbytes != [4]byte{0x0b, 0x11, 0x09, 0x07} {
-			fmt.Println("Option -testnet=true given but .dat file is NOT a testnet file.")
-			fmt.Println("Exiting...")
-			os.Exit(2)
-		}
-		f.Close()
-	} else {
-		f, err := os.Open("blk00000.dat")
-		if err != nil {
-			panic(err)
-		}
-		var magicbytes [4]byte
-		f.Read(magicbytes[:])
-
-		// Check if the magicbytes are for mainnet
-		if magicbytes != [4]byte{0xf9, 0xbe, 0xb4, 0xd9} {
-			fmt.Println("Option -testnet=true not given but .dat file is a testnet file.")
-			fmt.Println("Exiting...")
-			os.Exit(2)
+// is testnet3 or mainnet or regtest.
+func CheckNet(net wire.BitcoinNet) {
+	f, err := os.Open("blk00000.dat")
+	if err != nil {
+		panic(err)
+	}
+	defer f.Close()
+	var magicbytes [4]byte
+	f.Read(magicbytes[:])
+
+	bytesToMatch := make([]byte, 4)
+	// Can't use U32tB because we need little-endian
```

Comment:
In ```cmd/util/utils.go```, there is a function that does little-endian to u32.

I think it would be better if a ```U32tLB()``` was added and called here